Cargo.toml

```
@@ -31,8 +31,9 @@ default-features = false
 
 [dependencies.curve25519-dalek]
 package = "curve25519-dalek"
-version = "4.0.0-pre.2"
-default-features = false
+version = "4.0.0-rc.1"
+default-features = true
```

Right now this pulls in `alloc` and `precomputed-tables`, which will break HW wallets (there's no allocator on a HW wallet, and `precomputed-tables` dramatically expands the binary size which makes it too large to fit).

```suggestion
default-features = false
```

---

> Did we apply all the formatting fixes to this, or is that in upstream? If we did it, we should back out those changes and only make the ones which are actually required---otherwise future merges from master are going to include a ton of unnecessary conflicts.

Which formatting changes are you referring to? I'm not sure off the top of my head...

---

I removed `alloc` and the repo built just fine. However, when I remove `precomputed-tables`, the repo doesn't build. Here is an example error:

1) When we don't use `precomputed-tables`: 
```
error[E0425]: cannot find value `RISTRETTO_BASEPOINT_TABLE` in module `constants`
   --> src/keys.rs:569:54
    |
569 |         PublicKey::from_point(&self.key * constants::RISTRETTO_BASEPOINT_TABLE)
    |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `RISTRETTO_BASEPOINT_POINT`
```
I guess that the way forward is to make this code that uses `precomputed-tables` optionally compiled using `cfg_if`?

---

After looking into it, I don't think this code can be conditionally removed from the build. The issue is this constant `RISTRETTO_BASEPOINT_TABLE` only exists when `precomputed-tables` is enabled. Might we need to change the scalar multiplication logic here?

---

Ok, I removed `precomputed-tables` and replaced scalar multiplications that involved `RISTRETTO_BASEPOINT_TABLE` with `RISTRETTO_BASEPOINT_POINT`. The documentation on the differences between these two and how to use them is sparse, but it says: 
`The Ristretto basepoint, as a RistrettoPoint.
This is called _POINT to distinguish it from _TABLE, which provides fast scalar multiplication.`

This leads me to believe that both are interchangeable, the only difference is that the `_TABLE` is faster because it's precomputed. LMK if this sounds right...

---

> Which formatting changes are you referring to? I'm not sure off the top of my head...

The vast majority of changed lines in your commits look like rustfmt-on-save happened.

---

> After looking into it, I don't think this code can be conditionally removed from the build. The issue is this constant `RISTRETTO_BASEPOINT_TABLE` only exists when `precomputed-tables` is enabled. Might we need to change the scalar multiplication logic here?

The easy way to solve this is to put the compile-time configuration at the top of the file:

```rust
#[cfg(feature = "precomputed-tables")]
use curve25519_dalek::constants::RISTRETTO_BASEPOINT_TABLE as RISTRETTO_BASEPOINT;
#[cfg(not(feature = "precomputed-tables")]
use curve25519_dalek::constants::RISTRETTO_BASEPOINT_POINT as RISTRETTO_BASEPOINT;
```

---

If we want table or point depending on features perhaps it should be updated to use [`RistrettoPoint::mul_base()`](https://docs.rs/test-curve25519-dalek/4.0.0-rc.1/test_curve25519_dalek/ristretto/struct.RistrettoPoint.html#method.mul_base) as I'm guessing that's the intent of that function

---

Done.